### PR TITLE
feat: Optimize large file handling

### PR DESCRIPTION
**Pull request by JordanVoets**
## Description

Implements automatic chunking for large audio files to handle API file size limits gracefully. Files exceeding service-specific limits are automatically split into manageable chunks, transcribed separately, and results concatenated. Includes format-aware chunking with warnings for fragile formats like FLAC and Opus that may not resynchronize correctly at arbitrary byte boundaries.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [x] Code refactoring
- [x] Performance improvement
- [x] Test addition or modification

## Changes Made

- **Audio chunking utility** (`src/services/utils/audioChunker.ts`): Pure byte-based splitting without external dependencies, format-aware detection for robust vs. fragile formats, MP3 frame boundary detection for better chunk quality, configurable chunk sizes with safety margins
- **Chunked transcription wrapper** (`src/services/utils/chunkedTranscription.ts`): Transparent wrapper that automatically chunks large files, provides detailed progress logging, graceful error handling with context about which chunk failed
- **Main transcription flow** (`src/index.ts`): Replaced direct `service.transcribe()` calls with `transcribeWithChunking()` for automatic large file handling
- **Interface update** (`src/services/ITranscriptionService.ts`): Added required `getMaxFileSize()` method to interface, implemented in all service providers
- **Comprehensive test suite** (`tests/services/utils/audioChunker.test.ts`, `tests/services/utils/chunkedTranscription.test.ts`): 33 tests covering chunking logic, format detection, error handling, and edge cases
- **Extracted magic constants**: `SAFETY_MARGIN = 0.93` constant in tests for maintainability

## Testing

- [x] Tested with OpenAI Whisper provider (fixed 413 errors on 73.87 MB file by using correct safety margin)
- [x] Tested with Google Gemini provider (verified chunking works with base64 encoding overhead)
- [x] Tested with various audio formats (MP3, WAV, FLAC, Opus, OGG, AAC, M4A, WebM)
- [x] Manually tested in Joplin plugin environment (verified no FFmpeg dependency issues)
- [x] Added/updated unit tests (244 tests total, all passing)
- [x] All existing tests pass (13 test suites, 244 tests, 0 failures)

**Performance:**
- Full test suite runs in ~17 seconds
- audioChunker tests: ~1.3 seconds
- No memory issues (fixed ArrayBuffer allocations that were causing heap exhaustion)

## Checklist

- [x] My code follows the existing code style
- [x] I have commented my code where necessary
- [x] I have updated the documentation (if applicable)
- [x] My changes generate no new warnings (all linting passes)
- [x] I have tested my changes thoroughly (244 tests covering all scenarios)

## Additional Notes

**Key Design Decisions:**

1. **No external dependencies**: Uses pure Node.js Blob/ArrayBuffer APIs, works seamlessly in Joplin's sandboxed plugin environment without FFmpeg
2. **Format awareness**: Detects audio format fragility and warns users about formats (FLAC, Opus, OGG, AAC, M4A, WebM) that may produce unplayable chunks when split at arbitrary byte boundaries
3. **MP3 frame detection**: For MP3 files, attempts to find frame sync markers (`0xFFE`/`0xFFF`) to split at frame boundaries for better resilience
4. **Transparent to callers**: Chunking happens automatically inside the wrapper; existing code doesn't need to know about it
5. **Safety margins**: Applied 7-8% reduction to service limits to account for request encoding overhead (FormData/JSON)
6. **Sequential processing**: Chunks are transcribed sequentially to avoid API rate limiting issues

## Breaking Changes

**`ITranscriptionService` Interface Update:**
- Added required `getMaxFileSize(): number` method
- All service implementations must now implement this method

**Example for 73.87 MB file with OpenAI:**
- Before: 413 Content Too Large error
- After: Automatically split into 4 chunks (23 MB + 23 MB + 23 MB + 4.87 MB) → successful transcription

### CODE REVIEW

File: src/services/GeminiTranscriptionService.ts [hunk 1]
```diff
@@ -56,4 +56,11 @@ export class GeminiTranscriptionService implements ITranscriptionService {
       reader.readAsDataURL(blob);
     });
   }
+
+  getMaxFileSize(): number {
+    // Gemini API has a 100MB limit for base64-encoded payloads.
+    // Since base64 encoding increases size by ~33%, we limit the
+    // original file to ~75MB to stay within the API constraint.
+    return 75 * 1024 * 1024; // 75 MB
+  }
```
[the remark this answers is not in the document]
Comment (JordanVoets) [reply src/services/GeminiTranscriptionService.ts]: Removed it from docs. This overhead is now defined in the audio chunker.

**Comment by JordanVoets** [comment]
The current implementation chose to only support mp3s. Because of the way the audio is chunked in this implementation. It does not process other filetypes. More filetypes must be supported. WAV, M4A / AAC, MP3 and FLAC are the main filetypes that are used by different audio recorders so they must be supported at some point. This will require a redo of audioChunker.ts.

**Possible solutions**

1. Chunk the audio files using a package
2. Make audioChunkers for each filetype using an interface. This way, each filetype can have its own implementation

I prefer 1. Because 2 would require a lot more code, which is also code that must be maintained and makes the repo more complex.

**Comment by JordanVoets** [comment]
Made issue #28 based on my previous comment.
